### PR TITLE
Feature/bokeh plots

### PR DESCRIPTION
**Pull request by markbotterill**
Reviewer: @amiraliemami 
Estimate: 1-2 hours

---

## Ticket

Fixes: [JIRA_TICKET_LINK](https://idinsight.atlassian.net/jira/software/c/projects/AAQ/boards/43/backlog?selectedIssue=AAQ-788)

## Description

Adds topic modeling visualization to the front end

### Changes made

Topic modeling pipeline now writes to Redis during running
Front end pulls out this new Json stored in the Redis and renders interactive Bokeh with it

### Future Tasks (optional)

- Improve aesthetics -> CSS for buttons, update spacing within bokeh, make sure long text from hovertip does not overflow from plot
- Add a pop-up modal that allows for the creation of on-the-fly content cards (take logic from admin app main page and put into a smaller modal)
- Add a button for a "Soft refresh" -> Requires: Save BertTopic pipeline (and eliminate all stochasticity so that it can be reloaded, new content card can be updated and a new Bokeh plot can be rendered exactly the same only with new content embedding integrated. This will enable near-live plugging of content holes that can be verified on the fly

## How has this been tested?

Runs locally + passes tests

## Checklist

Fill with `x` for completed. 

- [ x] My code follows the style guidelines of this project
- [ x] I have reviewed my own code to ensure good quality
- [ x] I have tested the functionality of my code to ensure it works as intended
- [x ] I have resolved merge conflicts

(Delete any items below that are not relevant)
- [ ] I have updated the automated tests
- [ ] I have updated the scripts in `scripts/`
- [ ] I have updated the requirements
- [ ] I have updated the README file
- [ ] I have updated affected documentation
- [ ] I have added a blogpost in Latest Updates
- [ ] I have updated the CI/CD scripts in `.github/workflows/`
- [ ] I have updated the Terraform code

### CODE REVIEW

**Review by amiraliemami — COMMENTED** [review]
Leaving mid-way comments, thanks!!

**Review by amiraliemami — CHANGES_REQUESTED** [review]
Awesome work man, looks great!

Pls add the timeout required and reorder the components as below, thanks!

image

File: admin_app/src/app/dashboard/components/insights/Bokeh.tsx [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+          aria-controls="bokeh-plot-content"
+          id="bokeh-plot-header"
+          sx={{ width: "100%" }}
+        >
+          <Typography variant="h6">Topic Visualization</Typography>
+        </AccordionSummary>
+        <AccordionDetails>
+          {loadError && (
+            <Box sx={{ color: "error.main", mb: 2 }}>Error: {loadError}</Box>
+          )}
```
[the remark this answers is not in the document]
Comment (markbotterill) [reply admin_app/src/app/dashboard/components/insights/Bokeh.tsx:93]: @amiraliemami I have implemented a fix for this (see commit 95d606f) - let me know if you think it's sufficient

File: Makefile [hunk 2]
```diff
@@ -39,7 +39,6 @@ guard-%:
 
 # Add users to db
 add-users-to-db:
```
[the remark this answers is not in the document]
Comment (markbotterill) [reply Makefile:41]: Tried to do the whole branch w/o adding this as a commit 🤦 Must've slipped through - my bad.

File: core_backend/app/llm_call/llm_prompts.py [hunk 3]
```diff
@@ -410,7 +410,9 @@ class TopicModelLabellingResult(BaseModel):
     _prompt_base: str = textwrap.dedent(
         """
         You are a summarization bot designed to condense multiple
-        messages into a topic description specific to {context}. If unknown, respond
+        messages into a topic description specific to {context}. You may
+        encounter queries that are related to something other than maternal health.
```
[the remark this answers is not in the document]
Comment (markbotterill) [reply core_backend/app/llm_call/llm_prompts.py:414]: Used {context} again to avoid this loophole

**Comment by markbotterill** [comment]
> Awesome work man, looks great!
> 
> Pls add the timeout required and reorder the components as below, thanks!
> 
> image

@amiraliemami I added to the Dockerfile and removed the timeout command.

**Review by suzinyou — APPROVED** [review]
Works well. thank you Mark! This is amazing work

**Comment by suzinyou** [comment]
> Awesome work man, looks great!
> 
> Pls add the timeout required and reorder the components as below, thanks!
> 
> image

It looks like this now :) 

image

**Comment by markbotterill** [comment]
Thank you @suzinyou and @amiraliemami for all the feedback + help 🙏

**Pull request merged**